File: serial-write.cpp
#include "pxt.h"

using namespace pxt;

namespace serial {

//%
int writeStringBlocking(String text) {
    if (!text) {
        return MICROBIT_INVALID_PARAMETER;
    }
    return uBit.serial.send(MSTR(text), MicroBitSerialMode::SYNC_SPINWAIT);
}

//%
int writeBufferBlocking(Buffer buffer) {
    if (!buffer) {
        return MICROBIT_INVALID_PARAMETER;
    }
    return uBit.serial.send(buffer->data, buffer->length, MicroBitSerialMode::SYNC_SPINWAIT);
}

} // namespace serial

File: serial-write.ts
namespace serial {

    /**
    * Writes a text to the serial device. 
    */
    //% shim=serial::writeStringBlocking
    export function writeStringBlocking(text: string): number { return -1 }

    /**
    * Writes a buffer to the serial device. 
    */
    //% shim=serial::writeBufferBlocking
    export function writeBufferBlocking(buffer: Buffer): number { return -1 }
}